src/mbgl/style/expression/number_format.cpp

```diff
@@ -0,0 +1,222 @@
+#include <mbgl/style/expression/number_format.hpp>
+#include <mbgl/style/conversion_impl.hpp>
+#include <mbgl/util/platform.hpp>
+
+namespace mbgl {
+namespace style {
+namespace expression {
+
+const char localeKey[] = "locale";
+const char currencyKey[] = "currency";
+const char minFractionDigitsKey[] = "min-fraction-digits";
+const char maxFractionDigitsKey[] = "max-fraction-digits";
+
+NumberFormat::NumberFormat(std::unique_ptr<Expression> number_,
+                           optional<std::unique_ptr<Expression>> locale_,
+                           optional<std::unique_ptr<Expression>> currency_,
+                           optional<std::unique_ptr<Expression>> minFractionDigits_,
+                           optional<std::unique_ptr<Expression>> maxFractionDigits_)
```

Comment:
why not just `std::unique_ptr<Expression> maxFractionDigits_` , `maxFractionDigits(std::move(maxFractionDigits_));` (same for others) ?

Reply:
Done